prov/shm/src/smr_ep.c

```diff
@@ -467,6 +467,9 @@ static int smr_ep_close(struct fid *fid)
 	smr_recv_fs_free(ep->recv_fs);
 	smr_unexp_fs_free(ep->unexp_fs);
 	smr_pend_fs_free(ep->pend_fs);
+	smr_sar_fs_free(ep->sar_fs);
+	if (ep->name)
+		free((void *)ep->name);
```

Comment:
free accepts a null parameter, so we can skip the if check

Reply:
Thanks! I did not know that. Fixed in new revision